data/settings.js

```diff
@@ -3014,186 +3110,305 @@ If set, allows message deliveries to exceed quota by this value.`
+		seealso: [
+			'[[setting,quota_storage_extra]]',
+			'[[setting,quota_storage_percentage]]',
+			'[[setting,quota_message_count]]',
+			'[[link,quota_root]]'
+		],
+		tags: [ 'storage_size_limits' ],
+		text: `
+Quota storage size limit for the [[link,quota_root]]. This value is still
+multiplied by [[setting,quota_storage_percentage]] and then increased by
+[[setting,quota_storage_extra]] to get the final value (in this mailbox or
+namespace). This is reported as the STORAGE limit in IMAP GETQUOTA commands.
+Using \`0\` as the value means the same as \`unlimited\`.`
 	},
 
 	quota_warning: {
 		plugin: 'quota',
-		values: setting_types.STRING,
+		values: setting_types.NAMED_LIST_FILTER,
 		seealso: [ '[[link,quota_warning_scripts]]' ],
+		tags: [ 'quota-warning' ],
 		text: `
-You can configure Dovecot to run an external command when user's quota
-exceeds a specified limit. Note that the warning is ONLY executed at the
-exact time when the limit is being crossed, so when you're testing you have
-to do it by crossing the limit by saving a new mail. If something else
-besides Dovecot updates quota so that the limit is crossed, the warning is
-never executed.
-
-Quota warning configuration has the following syntax:
+Create a new [[link,quota_warning_scripts,quota warning]]. The filter name
+refers to [[setting,quota_warning_name]] setting. The [[setting,execute]]
+setting is required to be specified inside the filter.
 
-\`\`\`
-quota_warning = <limit configuration> <quota-warning socket name> <parameters>
-\`\`\`
-
-\`limit_configuration\` is almost exactly same as for
-[[setting,quota]], with the exception of adding \`-\` before
-the value for "reverse" warnings where the script is called when quota
-drops below the value. Usually you want to use percents instead of absolute
-limits.
+The order of \`quota_warning\` filters in the configuration is important:
+Only the first warning that matches the rules is executed. This means
+configuring the highest limits first.`
+	},
 
-Only the command for the first exceeded limit is executed, so configure the
-highest limit first. The actual commands that are run need to be created as
-services (create a named Dovecot service and use the service name
-as the \`quota-warning socket name\` argument).
+	quota_warning_name: {
+		plugin: 'quota',
+		values: setting_types.STRING,
+		seealso: [ '[[link,quota_warning_scripts]]' ],
+		tags: [ 'quota-warning' ],
+		text: `
+Name of the [[link,quota_warning_scripts,quota warning]]. The
+[[setting,quota_warning]] filter name refers to this setting. This name is
+only used within the configuration to identify the quota warning - it has no
+meaning otherwise.`
+	},
 
-Note: The percent sign (\`%\`) needs to be written as \`%%\` to avoid
-config expansion (see [[variable]]).
+	quota_warning_resource: {
+		plugin: 'quota',
+		values: setting_types.ENUM,
+		seealso: [ '[[link,quota_warning_scripts]]' ],
+		values_enum: [ 'storage', 'message' ],
```

Comment:
Maybe this would benefit a link to `[[link,quota_limits]]`. At least I was confused about what these mean until I scrolled way down to the quota page.